### DISCUSSION TITLE
Cron jobs not running the code.

### DISCUSSION THREAD

**Discussion by Rrhapsod** [vercel/vercel#10440] [Help] [answered]
### Question

So I've set a cron job on my app router "src/app/api/cron/route.ts" and created the schedule on the vercel.json.
After the deploy, I can run the cron route by other means, but if click Run on vercel dashboard, notting happens. On the logs, there is a sucess status code but the code inside the handler is not running. I have some Prisma updates like the example above:

```
import { NextResponse } from "next/server";
import { revalidatePath } from "next/cache";

import prisma from "@/lib/prisma.ts";

export async function GET() {

//some Prisma updates here

revalidatePath("/dashboard/financeiro/apagar/list");
revalidatePath("/dashboard/financeiro/areceber/list");
  return NextResponse.json({ ok: true });
}

```

**Comment by PlasticPrintersMN** [comment]
Did you find an answer to this?

**Comment by PlasticPrintersMN** [comment]
I'm running in to this exact situation. I started a discussion here: https://github.com/orgs/vercel/discussions/3903 on the org itself, then came over here and saw yours. I included a ton of info if you are interested in seeing if it's the same issue.

One thing I'd be interested in you checking out... when your cron runs, click on the 200 log and check the environment. Is it logging that it is running on a Preview build? I think this is what is causing the issue. I don't have any preview builds, therefor it is not actually running anything. But on your tests hitting the route through postman or browser etc. the log that works shows environment: Production. 

Curious if all of this is true for you. According to Vercel Docs, crons are supposed to be production only, so I have no clue what is happening, or how to fix it.

**Reply by Rrhapsod** [reply]
Yeah, the exact same thing. The cron jobs are all running on preview, I'm on Hobby Plan.

**Comment by Rrhapsod** [comment, marked as answer]
Just updating with the answer. Put this before your function.

`export const dynamic = 'force-dynamic'`
